task/upload.go

```diff
@@ -755,46 +769,35 @@ func outputLocations(
 	names, locations :=
 		[]OutputName{OutputNameOSPlaylistHLS, OutputNameEmpty},
 		[]clients.OutputLocation{
-			{
-				Type: "object_store",
-				URL:  url.JoinPath(hlsRelPath).String(),
-				Outputs: &clients.OutputsRequest{
-					HLS: hls,
-				},
-			},
-			{
-				Type: "object_store",
-				URL:  url.JoinPath(mp4RelPath).String(),
-				Outputs: &clients.OutputsRequest{
-					MP4: mp4,
-				},
-			},
+			outputLocation(url, hls.path, &clients.OutputsRequest{HLS: hls.enabled}),
+			outputLocation(url, mp4.path, &clients.OutputsRequest{MP4: mp4.enabled}),
 		}
-	if fmp4 == OUTPUT_ENABLED {
+	if fmp4.enabled == OUTPUT_ENABLED {
```

Comment:
Is this `enabled` field always `OUTPUT_ENABLED` or an empty string (or sth that means false)? Could be worth making it a boolean in the `output` struct to simplify, if thats the case

Reply:
An enum would be nice but then we'd still have to convert to the string for catalyst before sending the request.